src/librustdoc/doctest.rs

```diff
@@ -725,7 +732,17 @@ pub(crate) fn make_test(
         // /// ``` <- end of the inner main
         line_offset += 1;
 
-        prog.extend([&main_pre, everything_else, &main_post].iter().cloned());
+        // add extra 4 spaces for each line to offset the code block
+        let content = if opts.insert_extra_ident_space {
```

Comment:
Unless I missed something, it changes the generated test code and not just the playground URL right?

Reply:
I only set `opts.insert_extra_ident_space` to `true` for generating code link of Rust playground

Reply:
:+1: